### ISSUE TITLE
"Error: Cannot find tensor info for output.weight" on running llama 3.2 1B gguf

### ISSUE THREAD

**Issue by mathav95raj** [EricLBuehler/mistral.rs#798]
## Describe the bug
I am trying to run llama 3.2 1B gguf in Mac os (M3 chip). I am facing the above mentioned error with all available gguf models in hugging face. While gguf models of phi3.5 are running fine
## Latest commit or version
1eb9cae2a4ec89d7cf8a5fc8d9f57b82f2f747fa

```
cargo run --release -- -i gguf --quantized-model-id lmstudio-community/Llama-3.2-1B-Instruct-GGUF --quantized-filename Llama-3.2-1B-Instruct-Q4_K_M.gguf
    Finished `release` profile [optimized] target(s) in 0.39s
     Running `target/release/mistralrs-server -i gguf --quantized-model-id lmstudio-community/Llama-3.2-1B-Instruct-GGUF --quantized-filename Llama-3.2-1B-Instruct-Q4_K_M.gguf`
2024-09-27T11:01:21.693402Z  INFO mistralrs_server: avx: false, neon: true, simd128: false, f16c: false
2024-09-27T11:01:21.693478Z  INFO mistralrs_server: Sampling method: penalties -> temperature -> topk -> topp -> minp -> multinomial
2024-09-27T11:01:21.693547Z  INFO mistralrs_server: Model kind is: quantized from gguf (no adapters)
2024-09-27T11:01:21.693690Z  INFO hf_hub: Token file not found "/Users/391080/.cache/huggingface/token"    
2024-09-27T11:01:21.693742Z  INFO mistralrs_core::utils::tokens: Could not load token at "/Users/391080/.cache/huggingface/token", using no HF token.
2024-09-27T11:01:21.694175Z  INFO hf_hub: Token file not found "/Users/391080/.cache/huggingface/token"    
2024-09-27T11:01:21.694188Z  INFO mistralrs_core::utils::tokens: Could not load token at "/Users/391080/.cache/huggingface/token", using no HF token.
2024-09-27T11:01:23.605306Z  INFO mistralrs_core::pipeline::gguf: Loading model `lmstudio-community/Llama-3.2-1B-Instruct-GGUF` on cpu.
2024-09-27T11:01:23.968872Z  INFO mistralrs_core::gguf::content: Model config:
general.architecture: llama
general.basename: Llama-3.2
general.file_type: 15
general.finetune: Instruct
general.languages: en, de, fr, it, pt, hi, es, th
general.license: llama3.2
general.name: Llama 3.2 1B Instruct
general.quantization_version: 2
general.size_label: 1B
general.tags: facebook, meta, pytorch, llama, llama-3, text-generation
general.type: model
llama.attention.head_count: 32
llama.attention.head_count_kv: 8
llama.attention.key_length: 64
llama.attention.layer_norm_rms_epsilon: 0.00001
llama.attention.value_length: 64
llama.block_count: 16
llama.context_length: 131072
llama.embedding_length: 2048
llama.feed_forward_length: 8192
llama.rope.dimension_count: 64
llama.rope.freq_base: 500000
llama.vocab_size: 128256
2024-09-27T11:01:24.065246Z  INFO mistralrs_core::gguf::gguf_tokenizer: GGUF tokenizer model is `gpt2`, kind: `Bpe`, num tokens: 128256, num added tokens: 0, num merges: 280147, num scores: 0
2024-09-27T11:01:24.068291Z  INFO mistralrs_core::gguf::chat_template: Discovered and using GGUF chat template: `{% set loop_messages = messages %}{% for message in loop_messages %}{% set content = '<|start_header_id|>' + message['role'] + '<|end_header_id|> '+ message['content'] | trim + '<|eot_id|>' %}{% if loop.index0 == 0 %}{% set content = bos_token + content %}{% endif %}{{ content }}{% endfor %}{{ '<|start_header_id|>assistant<|end_header_id|> ' }}`
Error: Cannot find tensor info for output.weight
   0: std::backtrace::Backtrace::create
   1: candle_core::error::Error::bt
   2: mistralrs_core::gguf::content::Content<R>::tensor
   3: <mistralrs_core::models::quantized_llama::ModelWeights as mistralrs_core::utils::model_config::FromGGUF>::from_gguf
   4: mistralrs_core::utils::model_config::<impl core::convert::TryFrom<mistralrs_core::utils::model_config::ModelParams<mistralrs_core::utils::model_config::ParamsGGUF<R>>> for mistralrs_core::models::quantized_llama::ModelWeights>::try_from
   5: <mistralrs_core::pipeline::gguf::GGUFLoader as mistralrs_core::pipeline::loaders::Loader>::load_model_from_path
   6: <mistralrs_core::pipeline::gguf::GGUFLoader as mistralrs_core::pipeline::loaders::Loader>::load_model_from_hf
   7: mistralrs_server::main::{{closure}}
   8: tokio::runtime::park::CachedParkThread::block_on
   9: mistralrs_server::main
  10: std::sys::backtrace::__rust_begin_short_backtrace
  11: std::rt::lang_start::{{closure}}
  12: std::rt::lang_start_internal
  13: _main


Stack backtrace:
   0: std::backtrace::Backtrace::create
   1: anyhow::error::<impl core::convert::From<E> for anyhow::Error>::from
   2: <mistralrs_core::pipeline::gguf::GGUFLoader as mistralrs_core::pipeline::loaders::Loader>::load_model_from_path
   3: <mistralrs_core::pipeline::gguf::GGUFLoader as mistralrs_core::pipeline::loaders::Loader>::load_model_from_hf
   4: mistralrs_server::main::{{closure}}
   5: tokio::runtime::park::CachedParkThread::block_on
   6: mistralrs_server::main
   7: std::sys::backtrace::__rust_begin_short_backtrace
   8: std::rt::lang_start::{{closure}}
   9: std::rt::lang_start_internal
  10: _main
```

**Comment by swdunlop** [comment]
I see the same problem using the cpu device on MacOS and Linux for `bartowski/Llama-3.2-3B-Instruct-GGUF` and `bartowski/Llama-3.2-1B-Instruct-GGUF` using:

```rust
let model = GgufModelBuilder::new(
    "bartowski/Llama-3.2-3B-Instruct-GGUF",
    vec!["Llama-3.2-3B-Instruct-Q8_0.gguf"],
)
  .with_logging()
  .build()
  .await?;
```

**Comment by EricLBuehler** [comment]
Hi @swdunlop and @mathav95raj!

Thank you for the detailed error report. It looks like the Llama 3.2 GGUF quants used weight tying (using a `plain` model with ISQ would have worked), I have fixed the issue in #801!

```
cargo run --features cuda --release -- -i gguf -m lmstudio-community/Llama-3.2-1B-Instruct-GGUF -f  Llama-3.2-1B-Instruct-Q4_K_M.gguf
``` 

Please let me know if this works for you now!

**Comment by mathav95raj** [comment]
Thanks for the quick fix @EricLBuehler !

It is working with the same command now.

```
cargo run --release -- -i gguf --quantized-model-id lmstudio-community/Llama-3.2-1B-Instruct-GGUF --quantized-filename Llama-3.2-1B-Instruct-Q4_K_M.gguf
    Finished `release` profile [optimized] target(s) in 0.31s
     Running `target/release/mistralrs-server -i gguf --quantized-model-id lmstudio-community/Llama-3.2-1B-Instruct-GGUF --quantized-filename Llama-3.2-1B-Instruct-Q4_K_M.gguf`
2024-09-28T13:17:29.130354Z  INFO mistralrs_server: avx: false, neon: true, simd128: false, f16c: false
2024-09-28T13:17:29.130401Z  INFO mistralrs_server: Sampling method: penalties -> temperature -> topk -> topp -> minp -> multinomial
2024-09-28T13:17:29.130498Z  INFO mistralrs_server: Model kind is: quantized from gguf (no adapters)
2024-09-28T13:17:29.130618Z  INFO hf_hub: Token file not found "/Users/391080/.cache/huggingface/token"    
2024-09-28T13:17:29.130671Z  INFO mistralrs_core::utils::tokens: Could not load token at "/Users/391080/.cache/huggingface/token", using no HF token.
2024-09-28T13:17:29.131069Z  INFO hf_hub: Token file not found "/Users/391080/.cache/huggingface/token"    
2024-09-28T13:17:29.131077Z  INFO mistralrs_core::utils::tokens: Could not load token at "/Users/391080/.cache/huggingface/token", using no HF token.
2024-09-28T13:17:30.306904Z  INFO mistralrs_core::pipeline::gguf: Loading model `lmstudio-community/Llama-3.2-1B-Instruct-GGUF` on cpu.
2024-09-28T13:17:30.700553Z  INFO mistralrs_core::gguf::content: Model config:
general.architecture: llama
general.basename: Llama-3.2
general.file_type: 15
general.finetune: Instruct
general.languages: en, de, fr, it, pt, hi, es, th
general.license: llama3.2
general.name: Llama 3.2 1B Instruct
general.quantization_version: 2
general.size_label: 1B
general.tags: facebook, meta, pytorch, llama, llama-3, text-generation
general.type: model
llama.attention.head_count: 32
llama.attention.head_count_kv: 8
llama.attention.key_length: 64
llama.attention.layer_norm_rms_epsilon: 0.00001
llama.attention.value_length: 64
llama.block_count: 16
llama.context_length: 131072
llama.embedding_length: 2048
llama.feed_forward_length: 8192
llama.rope.dimension_count: 64
llama.rope.freq_base: 500000
llama.vocab_size: 128256
2024-09-28T13:17:30.796420Z  INFO mistralrs_core::gguf::gguf_tokenizer: GGUF tokenizer model is `gpt2`, kind: `Bpe`, num tokens: 128256, num added tokens: 0, num merges: 280147, num scores: 0
2024-09-28T13:17:30.799662Z  INFO mistralrs_core::gguf::chat_template: Discovered and using GGUF chat template: `{% set loop_messages = messages %}{% for message in loop_messages %}{% set content = '<|start_header_id|>' + message['role'] + '<|end_header_id|> '+ message['content'] | trim + '<|eot_id|>' %}{% if loop.index0 == 0 %}{% set content = bos_token + content %}{% endif %}{{ content }}{% endfor %}{{ '<|start_header_id|>assistant<|end_header_id|> ' }}`
2024-09-28T13:17:32.161243Z  INFO mistralrs_core::pipeline::paths: Using literal chat template.
2024-09-28T13:17:32.280775Z  INFO mistralrs_core::pipeline::chat_template: bos_toks = "<|begin_of_text|>", eos_toks = "<|eot_id|>", unk_tok = `None`
2024-09-28T13:17:32.284221Z  INFO mistralrs_server: Model loaded.
2024-09-28T13:17:32.284328Z  INFO mistralrs_server::interactive_mode: Starting interactive loop with sampling params: SamplingParams { temperature: Some(0.1), top_k: Some(32), top_p: Some(0.1), min_p: Some(0.05), top_n_logprobs: 0, frequency_penalty: Some(0.1), presence_penalty: Some(0.1), stop_toks: None, max_len: Some(4096), logits_bias: None, n_choices: 1, dry_params: Some(DrySamplingParams { sequence_breakers: ["\n", ":", "\"", "*"], multiplier: 0.0, base: 1.75, allowed_length: 2 }) }
> write a poem filled with modern day puns on language models
"The Model of the World"  by Emily Dickinson

In silicon halls, where data reigns
A language model reigns, with algorithms' pains
It speaks in tongues, of syntax and flair
But sometimes, its humor's beyond compare

It's a "text-ual" nightmare, to decipher its intent
A "word-ful" of errors, that make our hearts invent
It's a "language" of logic, that's hard to define
But when it "tells" us something, we're left in a "state" of mind

It's a "model" of the world, that's full of bias and might
A "language" of the masses, that's lost in the night
It's a "text-ural" puzzle, that's hard to solve
But when it "speaks" to us, we're left with a "story" to evolve

In this world of code and circuitry
```
